src/radix_connect/mobile/client.rs

```diff
@@ -1,49 +1,174 @@
-// use super::relay_service::Service as RelayService;
+use std::borrow::BorrowMut;
+
 use super::deep_link_parsing::*;
+use super::relay_service::Service as RelayService;
 use crate::prelude::*;
+use std::sync::RwLock;
 
 /// The Radix Connect Mobile client.
 /// This is the object that will be used by the mobile app to handle interactions sent over Radix Connect Relay.
-// #[derive(uniffi::Object)]
-// pub struct RadixConnectMobile {}
-
-// // Provisional API
-// #[uniffi::export]
-// impl RadixConnectMobile {
-//     // RadixConnectMobile should require a NetworkAntenna and a SecureStorage from the Wallet.
-//     // The internal components, such as RadixConnectRelayService will be created by the RadixConnectMobile.
-//     #[uniffi::constructor]
-//     pub fn new(
-//         _network_antenna: Arc<dyn NetworkAntenna>,
-//         _secure_storage: Arc<dyn SecureStorage>,
-//     ) -> Self {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn handle_linking_request(
-//         &self,
-//         _request: RadixConnectMobileLinkRequest,
-//     ) -> Result<Url> {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn handle_dapp_interaction_request(
-//         &self,
-//         _request: RadixConnectMobileDappRequest,
-//     ) -> Result<DappToWalletInteraction> {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn send_dapp_interaction_response(
-//         &self,
-//         _response: WalletToDappInteractionResponse,
-//     ) -> Result<Url> {
-//         todo!()
-//     }
-// }
+#[derive(uniffi::Object)]
+pub struct RadixConnectMobile {
+    /// The Radix Connect Relay service to be used to communicate with dApps.
+    relay_service: RelayService,
+    /// The secure storage to be used to store session data.
+    secure_storage: WalletClientStorage,
+    /// The new sessions that have been created and are waiting to be validated on dApp side.
+    /// Once the session is validated, it will be moved to the secure storage.
+    /// Validation consists in verifying the origin of the session.
+    new_sessions: RwLock<HashMap<SessionID, Session>>,
```

Comment:
I think a `pub type InFlightSession = Session;` and -> `in_flight_sessions: RwLock<HashMap<SessionID, InFlightSessions>>` 

or maybe "In Flight" is not best... maybe `PendingLinkingCompletedSession` and `pending_linking_completed_sessionss` etc?

Reply:
see related comment https://github.com/radixdlt/sargon/pull/116/files#r1584715414